lib/http/features/auto_inflate.rb

```diff
@@ -17,15 +17,15 @@ def wrap_response(response)
           :headers       => response.headers,
           :proxy_headers => response.proxy_headers,
           :connection    => response.connection,
-          :body          => stream_for(response.connection),
+          :body          => stream_for(response.connection, :encoding => response.body.encoding),
           :request       => response.request
         }
 
         Response.new(options)
       end
 
-      def stream_for(connection)
-        Response::Body.new(Response::Inflater.new(connection))
+      def stream_for(connection, encoding: Encoding::BINARY)
+        Response::Body.new(Response::Inflater.new(connection), :encoding => encoding)
```

Comment:
`stream_for` can and should be private, and `encoding:` seems like don't need default value.